### ISSUE TITLE
adjust the learning rate

### ISSUE THREAD

**Issue by pepetko80** [Microsoft/CNTK#2294]
Currently it is not possible to auto-adjust the learning rate when using the Python interface. Is there a work-around? So, can I access the current "lr" value at the end of an epoch and modify it based on some criterion, e.g. minimal improvement?
Thank you!

**Comment by ebarsoumMS** [comment]
Yes, you can call `reset_learning_rate` on the learner anytime during training.

https://cntk.ai/pythondocs/cntk.learners.html?highlight=reset_learning_rate#cntk.learners.Learner.reset_learning_rate

**Comment by pepetko80** [comment]
Thank you for the response! I am trying to make it work but encounter a problem. I can access the learning rate through the "learner" but not through the "trainer" designed based on the "learner". Does that mean that I have to re-create the "trainer" and then reload the state from the last epoch. Is it possible to modify the learning rate via the "trainer"?

**Comment by ebarsoumMS** [comment]
No, you can keep a reference to the learner that you past to the trainer, ad use that learner.

**Comment by pepetko80** [comment]
Thank you Emad, it worked out! With this in place "autoadjust"-ing the learning rate becomes trivial.
